engines/extpoints/interfaces.go

```diff
@@ -32,4 +32,4 @@ type EngineOptions struct {
 // Any error here will be fatal and likely cause the worker to stop working.
 // If an implementor can determine that the platform isn't supported at
 // compile-time it is recommended to not register the implementation.
-type EngineProvider func(options EngineOptions) (engines.Engine, error)
+type EngineProvider func(options *EngineOptions) (engines.Engine, error)
```

Comment:
Ok, I can change it back to not passing around a pointer.  Changing between a pointer and a value didn't change the behavior we're trying to accomplish though....passing around a struct that can have fields that change without needing to change the method signature of the receivers of it.